lib/plugins/time.js

```diff
@@ -2,20 +2,45 @@ module.exports = inject
 
 function inject (bot) {
   bot.time = {
+    doDaylightCycle: null,
+    time: null,
+    timeOfDay: null,
     day: null,
+    isDay: null,
+    moonPhase: null,
     age: null
   }
   bot._client.on('update_time', (packet) => {
-    bot.time.time = longToNumber(packet.time)
+    let time = longToNumber(packet.time)
+
+    if (time < 0) {
+      bot.time.doDaylightCycle = false
+      time *= -1
+    } else {
+      bot.time.doDaylightCycle = true
+    }
+
+    bot.time.time = time
     bot.time.timeOfDay = bot.time.time % 24000
     bot.time.day = Math.floor(bot.time.time / 24000)
     bot.time.isDay = bot.time.timeOfDay < 13000 || bot.time.timeOfDay >= 23000
     bot.time.moonPhase = bot.time.day % 8
     bot.time.age = longToNumber(packet.age)
+
     bot.emit('time')
   })
 }
 
 function longToNumber (arr) {
-  return arr[1] + 4294967296 * arr[0]
+  const str = (arr[0] >>> 0).toString(2) + (arr[1] >>> 0).toString(2)
+  if (str[0] === '1') {
+    // If the first character is a 1, it means the number is negative.
+    // To parse the number we first need to split the string to an array.
+    // Then we can flip each bit, and join it to a string again.
+    // Then the string is parsed to an int using base 2.
+    // And lastly the result is negated and subtracted by 1.
+    return parseInt(str.split('').map(bit => 1 - bit).join(''), 2) * -1 - 1
```

Comment:
Just a suggestion:
```js
return -1 - parseInt(str, 2) ^ 2**str.length - 1
```

Reply:
Hmm, it doesn't seem to work correctly.

I'm starting to think that `parseInt` somehow doesn't work bit such large numbers?  
When I do `console.log(parseInt('1111111111111111111111111111111111111111111111111111111111111111', 2))` (64 1s) I should just get 2^64 right?  
But instead I get 18446744073709552000 which is `18446744073709552000 - 2^64 = 384` off.

2^64 = 18446744073709551616, so the last 3 digits are rounded up it seems.

Reply:
That's because the largest representable integer in js is `2^53-1` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

Reply:
Welp, that makes sense then.

So I guess the best way to parse the number is to just flip it and parse it as a positive number first, like it is now?  
That way the inaccuracy at least doesn't show up with smaller numbers.

Reply:
The ideal way to fix issues like this I think is to read 64bit integers into BigInts in protodef :)

In order to know if the number is negative, all you need is the upper half of the integer, you don’t need to go through the process of merging them. If you do want to merge them, do it the right way and use a BigInt.

Reply:
untested, but something like this should be able to read it as a signed BigInt: 
`BigInt.asIntN(64, (BigInt(upper) << 32n)) | BigInt(lower)`

Reply:
Using a BigInt seems to works, but it does mean that all the values (bot.time.time etc) also need to be a BigInt, as converting it to a normal number wouldn't make sense right?  
I don't know whether using a BigInt for those values is a good thing or not?

Reply:
This would be an API break, but IMO if it’s unlikely that the value will ever get that high into the 64 bit integer other than negative numbers, it’s not really necessary for the breaking change. Because of the way JS works, numbers and BigInts cannot interact with each other without being converted. Maybe an additional .tickTime or something similar could be added to store the precise tick time if the numbers can get really big.

Reply:
Yeah that would seem like the most sensible option.

I've implemented the changes and also added something to the docs to explain when to use which value.

Let me know if it needs some improvements.